test/spec/Feature/Auth/JwtCacheSpec.hs

```diff
@@ -0,0 +1,22 @@
+module Feature.Auth.JwtCacheSpec where
+
+import Network.Wai (Application)
+
+import Test.Hspec
+import Test.Hspec.Wai
+import Test.Hspec.Wai.JSON
+
+import Protolude hiding (get)
+
+spec :: SpecWith ((), Application)
+spec = describe "jwt cache enabled" $ do
+  it "denies access to tables that anonymous does not own" $
+    get "/authors_only" `shouldRespondWith`
+      [json| {
+        "hint":null,
+        "details":null,
+        "code":"42501",
+        "message":"permission denied for table authors_only"} |]
+      { matchStatus = 401
+      , matchHeaders = ["WWW-Authenticate" <:> "Bearer"]
+      }
```

Comment:
I don't understand the purpose of this test on this PR. Isn't this already tested here?

https://github.com/PostgREST/postgrest/blob/d0173e71923e15b412fd459e35e1d5b97161a363/test/spec/Feature/Auth/AuthSpec.hs#L17-L26

Is it a way to test the code path of the JWT cache being enabled and that this permission error happens?

I think what we need to test here is that the JWT cache metric doesn't increase when we add a new empty JWT. And that should be an io test. The permission error message can be tested there too.

Reply:
> Is it a way to test the code path of the JWT cache being enabled and that this permission error happens?

Yes , I mainly added this for code-coverage reasons. Hmm, this PR is only for the empty JWT, and the metric is on the #3802, so we can't test that here.